pkg/promql/common.go

```diff
@@ -60,6 +60,22 @@ func MinBy(metricName string, byLabels []string, labelMatchers ...*labels.Matche
 	).By(byLabels...)
 }
 
+func AvgBy(metricName string, byLabels []string, labelMatchers ...*labels.Matcher) parser.Expr {
+	return promqlbuilder.Avg(
+		matrix.New(
+			vector.New(vector.WithMetricName(metricName), vector.WithLabelMatchers(labelMatchers...)),
+		),
```

Comment:
So the 0s we see above is due to matrix input.

Maybe for avg by, max by, min by we add the matrix.WithRangeAsVariable here?